src/include/brain/catalog_sync_brain_job.h

```diff
@@ -0,0 +1,124 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// catalog_sync_brain_job.h
+//
+// Identification: src/include/brain/catalog_sync_brain_job.h
+//
+// Copyright (c) 2015-2018, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+#pragma once
+#include <vector>
+#include <string>
+#include <pqxx/pqxx>
+#include <algorithm>
+#include "concurrency/transaction_manager_factory.h"
+#include "brain/brain.h"
+#include "catalog/catalog.h"
+#include "catalog/table_catalog.h"
+#include "storage/data_table.h"
+#include "catalog/schema.h"
+#include "type/value_factory.h"
+#include "catalog/query_history_catalog.h"
+#include "catalog/query_metrics_catalog.h"
+
+namespace peloton {
+namespace brain {
+class CatalogSyncBrainJob : public BrainJob {
```

Comment:
This class overall looks good! Two small issues:
1. Some brief comments would be good.
2. I would suggest moving the definitions of all the functions except `CatalogSyncBrainJob(BrainEnvironment *env)` into the cpp file and reduce includes as much as possible.

It is a hacky way to do syncing, but it will stay here for a while...